breathecode/registry/tests/tasks/tests_async_download_single_readme_image.py

```diff
@@ -0,0 +1,170 @@
+                    bucket=PropertyMock(),
+                    file_name=PropertyMock(),
+                    blob=PropertyMock(side_effect=[None, 1]),
+                    upload=MagicMock(),
+                    url=MagicMock(return_value='https://xyz/hardcoded_url'),
+                    create=True)
+    @patch('os.getenv',
+           MagicMock(side_effect=apply_get_env({
+               'GOOGLE_PROJECT_ID': 'labor-day-story',
+               'MEDIA_GALLERY_BUCKET': 'bucket-name',
+           })))
+    def test__with_download_status_no_asset_image(self):
+        model = self.bc.database.create(asset={'slug': 'fake_slug'})
+
+        result = async_download_single_readme_image('fake_slug', original_url)
+
+        readme = self.bc.database.get_model('registry.asset').objects.first().get_readme()['decoded']
+        asset_image = self.bc.database.get_model('registry.AssetImage').objects.first()
+        self.assertEqual(result, 'OK')
+        self.assertEqual('https://xyz/hardcoded_url' in readme, False)
+        self.assertEqual(self.bc.database.list_of('registry.AssetImage'), [{
+            'id': 1,
+            'bucket_url': 'https://xyz/hardcoded_url',
+            'original_url': original_url,
+            'download_details': f'Downloading {original_url}',
+            'download_status': 'OK',
+            'hash': asset_image.hash,
+            'last_download_at': asset_image.last_download_at,
+            'mime': 'image/png',
+            'name': 'www.google.com',
+        }])
+
+    @patch('requests.get',
+           apply_requests_get_mock([(200, original_url, {
+               'headers': {
+                   'content-type': 'image/png'
+               }
+           }, {
+               'content-type': 'image/png'
+           })]))
+    @patch.multiple('breathecode.services.google_cloud.Storage',
+                    __init__=MagicMock(return_value=None),
+                    client=PropertyMock(),
+                    create=True)
+    @patch.multiple('breathecode.services.google_cloud.File',
+                    __init__=MagicMock(return_value=None),
+                    bucket=PropertyMock(),
+                    file_name=PropertyMock(),
+                    blob=PropertyMock(side_effect=[None, 1]),
+                    upload=MagicMock(),
+                    url=MagicMock(return_value='https://xyz/hardcoded_url'),
+                    create=True)
+    @patch('os.getenv',
+           MagicMock(side_effect=apply_get_env({
+               'GOOGLE_PROJECT_ID': 'labor-day-story',
+               'MEDIA_GALLERY_BUCKET': 'bucket-name',
+           })))
+    def test__with_download_status_not_ok(self):
+        asset_image = {'name': 'john', 'original_url': original_url, 'bucket_url': 'https://www.f.com'}
+        model = self.bc.database.create(asset={'slug': 'fake_slug'}, asset_image=asset_image)
+
+        readme = model['asset'].get_readme()
+        model['asset'].set_readme(readme['decoded'] + ' https://www.f.com')
```

Comment:
Should be better to have the content hardcoded instead of call `get_readme()` and `set_readme()` because a change of those behaviors should broke the behaviors of this tests